src/nvim/tui/tui.c

```diff
@@ -1585,6 +1588,11 @@ static void patch_terminfo_bugs(TUIData *data, const char *term,
       unibi_set_if_empty(ut, unibi_set_left_margin_parm, "\x1b[%i%p1%ds");
       unibi_set_if_empty(ut, unibi_set_right_margin_parm, "\x1b[%i;%p2%ds");
     }
+
+#ifdef WIN32
+    // Replace cud1 because \n is CRLF on Windows.
+    unibi_set_str(ut, unibi_cursor_down, "\x1b[B");
```

Comment:
Why in `if (xterm)` ? Shouldn't it be done always on Windows?

Reply:
The `cud1` of the non-xterm terminal may not be `\x1b[B`. I think we have to add `cud1` conversion each time we find a terminal that causes a problem.

I think the fundamental solution is to implement a mode that does not convert `LF` to `CRLF` in `libuv`. If we write a patch, it may not be taken upstream, what shuld we do?

Reply:
> I think the fundamental solution is to implement a mode that does not convert `LF` to `CRLF` in `libuv`. If we write a patch, it may not be taken upstream, what shuld we do?

If it's a small/low-risk patch, they will probably accept it.

Reply:
@erw7 yeah, I think the main issue here is the automatic conversion of `LF` to `CR``LF`. "Proper" terminal emulators can deal with both and I know Alacritty supports the "legacy newline mode" too if needed.

Reply:
It seems that `libuvu` is not the only one that converts `CR` to `CR+LF`. It also seems to occur when `uv__vterm_state == UV_SUPPORTED`. So it seems to me that `ConEmu` and `VTP` are doing the same conversion.

Therefore, in order to solve this problem on Windows, it may be best to ignore the `TERM` environment variable even if it is set.